course_content/notebooks/dask_intro.ipynb

```diff
@@ -393,6 +393,281 @@
+    "        \n",
+    "        iris.cube.CubeList(db.from_sequence(files).map(iris.load_cube).compute())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load multiple files into single cube using `iris.load`\n",
+    "If the data you are loading from multiple files represents the same phenomenon then when the data is stored as multiple cubes can be merged into a single cube. For instance from our previous example we have 120 cubes each representing the northward sea ice velocity in a single month and collectively representing that data for the decade 1890-1899. We can merge this dataset into a single cube using Iris as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "merged_cube = iris.load(files).merge_cube()\n",
+    "\n",
+    "print(merge_cube)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Compare this to the ouput of printing a single cube and it is clear that the decade's worth of data has been merged into a single cube with 120 time steps:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "single_cube = iris.load_cube(files[0])\n",
+    "\n",
+    "print(single_cube)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Load multiple files into single cube using `dask.bag` and `dask.delayed`\n",
+    "We can merge the `CubeList` we generate using a `dask.bag` in exactly the same way:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "bag_merged_cube = bag_cubelist.merge_cube()\n",
+    "\n",
+    "print(bag_merged_cube)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "However the `merge_cube` step in this does not leverage Dask. We can utilise Dask in the merge step by decorating a call to `merge_cube` with the `@delayed` decorator. This allows us to integrate it with our task graph:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@delayed\n",
+    "def del_merge(cubes):\n",
+    "    return iris.cube.CubeList(cubes).merge_cube()\n",
+    "\n",
+    "del_merged_cube = del_merge(bag_o_cubes)\n",
+    "\n",
+    "del_merged_cube.visualize()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once again if you expand the task graph (double-click image) you can see that `merge_cube()` is now part of it. Compare this to the task graph of `bag-o-cubes`. What is different?\n",
```

Comment:
Nice – a mini-exercise embedded in the main content. Could we perhaps add a blank cell below for the learner to add their answer?

Reply:
#111 addresses :white_check_mark: